### PR TITLE
Add webteam practices as discourse docs

### LINKED ISSUE #96: SSO is causing a redirect error on /guides
- Go to https://webteam.canonical.com/
- Click "Guides" in the top nav
- You'll see an error stating that the page tried to redirect you too many times

This doesn't happen locally, instead it takes you to the SSO login page, and redirects you to /guides as expected. It's also affecting https://github.com/canonical-web-and-design/webteam.canonical.com/pull/93.

### PR DESCRIPTION
**Pull request by jpmartinspt**
Add a practices section powered by discourse.

Fixes https://github.com/canonical-web-and-design/webteam.canonical.com/issues/91
Fixes #96 

# QA
- set the discourse credentials in `.env.local`
- dotrun
- got to `/practices`
- navigate a bit and check it works
- fiddle with the index topic to improve anything you want

### CODE REVIEW

File: webapp/guides.py
Comment (jpmartinspt) [inline webapp/guides.py:29]: I was not entirely sure we needed this one, I think we don't because we already do the same [here](https://github.com/canonical-web-and-design/webteam.canonical.com/pull/94/files#diff-8235c5f219f2fb4d725e22f674ae43d399f5464fa6e493af349e0d0641401bbcR54)

File: webapp/sso.py
Comment (nottrobin) [inline webapp/sso.py:51]: This file was created by @tbille, so I guess I'll address this question to both of you:

In our other apps we have an `authorization_required()` decorator that we put above views. Do you know why we don't do this here? This seems to split up the concerns in a way that's unnecessarily surprising.

Comment (nottrobin) [reply webapp/sso.py:51]: Oh ok now I see - the view is created by the discourse module and so it's hard to wrap the authorization around it... OK I guess this is alright for now. If I get time I'll see if I can think of a better structure.

File: webapp/sso.py
Comment (nottrobin) [inline webapp/sso.py:55]: @anthonydillon are we cool with adding this pack behind SSO for now? Or should we hold off entirely until we've decided what to make public?

**Review by nottrobin — DISMISSED** [review]
Got this when trying to QA practices:

image

File: webapp/sso.py
Comment (anthonydillon) [reply webapp/sso.py:55]: Yes behind SSO for now but we should be book time in soon to unblock this content

**Review by nottrobin — CHANGES_REQUESTED** [review]
Please just make the /practices area public. Then good to go. I think.

Oh and rebase. Branch out of date.

**Comment by cristinadresch** [comment]
@jpmartinspt could you have a look at this one?

**Comment by cristinadresch** [comment]
@nottrobin can you have a look again pls

**Comment by nottrobin** [comment]
What's concerning me about this is, will it still have the login redirect loop problem we have on live? https://webteam.canonical.com/guides

**Comment by nottrobin** [comment]
Seems like it does:

https://webteam-canonical-com-94.demos.haus/

image

**Comment by jpmartinspt** [comment]
@nottrobin it should all be working now.

**Comment by sowasred2012** [comment]
QA looks good to me! No redirect loops at least - I don't feel confident reviewing the code, maybe @nottrobin ?

File: webapp/sso.py
Comment (nottrobin) [inline webapp/sso.py:7]: Was this the fix to the infinite redirect? I'd say we definitely want this to be at least company public. If it's the difference between `canonical` and `canonical-webmonkeys` (i.e. the difference between public and private launchpad teams) that make this work, then we should talk to @Roadmaster about getting our sites permission to use the private `canonical` group.

Comment (jpmartinspt) [reply webapp/sso.py:7]: No the issue with the redirect, was missing dependencies in the container. Using the `canonical` group does not work at the moment and `canonical-webmonkeys` sounded like a reasonable compromise as everyone in the team is usually added to it.

Comment (jpmartinspt) [reply webapp/sso.py:7]: If @nottrobin is happy, feel free to press the merge button. I think the other issue about the `canonical` group can be tackled separately.

**Review by anthonydillon — APPROVED** [review]
LGTM :+1:

**Pull request merged**
